MOTR-23 MOTR-22 Integration of CreateReactApp based Front-end

akre96 opened the pull request:
Includes components related to file uploading, landing page, initial dashboard. Includes the first batch of tests and snapshot tests. Integrated with Storybook.

kilodalton commented on public/index.html:
```diff
@@ -0,0 +1,42 @@
+    <link rel="shortcut icon" href="%PUBLIC_URL%/favicon.ico">
+    <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
+    <link href="/open-iconic/font/css/open-iconic-bootstrap.css" rel="stylesheet">
+    <link href="%PUBLIC_URL/main.css" rel="stylesheet">
```
@akre96 you are missing a `%` here, which generates a `URIError: Failed to decode param '/%PUBLIC_URL/main.css'` error:

```suggestion
    <link href="%PUBLIC_URL%/main.css" rel="stylesheet">
```

kilodalton commented:
@akre96 How many icons are used from the open-iconic set? A better practice is to include and only version control those.

jimmyzhen commented on package.json:
```diff
@@ -0,0 +1,50 @@
+    "build-storybook": "build-storybook -s public",
+    "sass": "sass --watch src/sass/main.scss src/main.css"
+  },
+  "eslintConfig": {
```
This field is redundant in `package.json` if we are using `.eslintrc.js`.

akre96 replied:
Right, that makes sense, it is removed now.

jimmyzhen commented on package.json:
```diff
@@ -0,0 +1,50 @@
+    "@storybook/react": "^3.4.11",
+    "bootstrap": "^4.1.3",
+    "enzyme": "^3.7.0",
+    "eslint-config-airbnb": "^17.1.0",
```
Unit testing and linting pacakges are preferred to be installed as `devDependencies`.

akre96 replied:
Done!  ESlint gives an error stating storybook should be in dependencies and not devDependencies, but I agree it should be in devDependencies.

jimmyzhen commented on src/App.jsx:
```diff
@@ -0,0 +1,36 @@
+import dashboard from './components/dashboard';
+import uploadScreen from './components/uploadScreen';
+
+require('bootstrap');
```
Alternatively, we can use `import 'bootstrap';`

See below for reference:
https://getbootstrap.com/docs/4.0/getting-started/webpack/#importing-javascript

jimmyzhen commented on src/components/footer.jsx:
```diff
@@ -0,0 +1,80 @@
+  onLogIn,
+  onLogOut,
+}) {
+  if (loggedInStatus === true) {
```
When evaluating a boolean, a shorthand (e.g. `if (loggedInStatus)`) can be used as well.
```suggestion
  if (loggedInStatus) {
```

akre96 replied:
Implemented! This was initially there because of the other error you pointed out in defaultAuthProps where false was passed as a string type, meaning without the strict equality it would have passed.

jimmyzhen commented on src/components/dashboard.jsx:
```diff
@@ -0,0 +1,36 @@
+    <div className="container">
+      <div className="row">
+        <div className="col">
+          <br />
```
Using CSS to control vertical spacing is better than using `<br />` tags.

akre96 replied:
Removed several <br /> tags in current PR. Current remaining ones are primarily to split the text into two lines. Will continue to scan and reimplement spacing without <br /> tags as I see them in future PRs.

jimmyzhen commented on src/components/footer.jsx:
```diff
@@ -0,0 +1,80 @@
+            </p>
+          </div>
+          <div className="col copyright">
+            <p>&#169; XXXX 2018</p>
```
If the copyright info is actually needed, then I would suggest avoiding the use of static _year_.
```suggestion
            <p>&#169; XXXX {getCopyrightYear()}</p>
```
```
function getCopyrightYear() {
  const today = new Date();
  const year = today.getFullYear();
  return year;
}
```

akre96 replied:
Implemented! This makes a lot of sense.

jimmyzhen commented on src/components/uploadAreaDnD.jsx:
```diff
@@ -0,0 +1,51 @@
+        <br />
+        <br />
+        <br />
+        <br />
```
Using CSS to control vertical spacing is more preferable than `<br />` tags.

akre96 replied:
implemented! Will continue to monitor for this and change in future PRs as well.

jimmyzhen commented on src/reducers/authReducer.js:
```diff
@@ -0,0 +1,25 @@
+export const defaultAuthState = {
+  user: {},
+  loggedIn: 'false',
```
Is this intended to be a _string_ type?

akre96 replied:
Not at all, thanks for catching that. Helped clean a lot of the issues I was having.

akre96 commented:
**Changes implemented:** 
- Fixed %PUBLIC_URL% relative link issue
- Removed icon files except for font section
- updated package.json to remove redundancies, esLintConfig, and moved relevant items to devDependenices
- Fixed "defaultAuthState" bug where 'false' was being returned as string instead of boolean value
- Simplified checking LoggedIn logic
- Copyright year returned from a function instead of a static value
- Compiled snapshot and css files removed
- .gitIgnore updated to include compiled files
- removed several <br /> and replaced with CSS stylings, the remainder will be fixed in subsequent PRs
- importing bootstrap javascript will be changed to "import 'bootstrap';" in future PRs

kilodalton commented:
Merged MOTR-23 MOTR-22

Strong work, @akre96 ! 🎉
